File: C/EX_estrutura_sequencial/ex3_idades/idades.c
/*
Exemplo: 
Dados da primeira pessoa: 
Nome: Maria Silva
Idade: 19
Dados da segunda pessoa: 
Nome: Joao Melo
Idade 20
A idade média de Maria Silva e Joao Melo é de 19.5 anos*/

#include<stdio.h>
#include<string.h>

void ler_texto(char *buffer, int length) { 
 fgets(buffer, length, stdin); 
 strtok(buffer, "\n"); 
} 
void limpar_entrada() { 
 char c; 
 while ((c = getchar()) != '\n' && c != EOF) {} 
}

int main()
{
    char nome1[50];
    int idade1;
    char nome2[50];
    int idade2;
    double media_idade;

    printf("Dados da primeira pessoa:\n ");
    printf("\nNome: ");
    ler_texto(nome1, 50);
    printf("Idade: ");
    scanf("%d", &idade1);

    printf("\n");

    printf("Dados da segunda pessoa:\n ");
    printf("\nNome: ");
    limpar_entrada();
    ler_texto(nome2, 50);
    printf("Idade: ");
    scanf("%d", &idade2);

    media_idade = (double)(idade1 + idade2) / 2;

    printf("A idade media de %s e %s eh de %.1lf anos", nome1, nome2, media_idade);


    return 0;
}

File: Java/Ex_estrutura_sequencial/ex2 idades/idades.java
/*
Exemplo: 
Dados da primeira pessoa: 
Nome: Maria Silva
Idade: 19
Dados da segunda pessoa: 
Nome: Joao Melo
Idade 20
A idade média de Maria Silva e Joao Melo é de 19.5 anos 
 */

import java.util.Scanner; 
import java.util.Locale;  

public class idades {
    public static void main (String[] args) {
        
        Locale.setDefault(Locale.US); 
        Scanner sc = new Scanner(System.in); 

        String nome1, nome2;
        int idade1, idade2;
        double media_idade;

        System.out.println("Dados da primeira pessoa:");
        System.out.print("Nome: ");
        nome1 = sc.nextLine();
        System.out.print("Idade: ");
        idade1 = sc.nextInt();

        System.out.println("Dados da segunda pessoa:");
        System.out.print("Nome: ");
        sc.nextLine();
        nome2 = sc.nextLine();
        System.out.print("Idade: ");
        idade2 = sc.nextInt();

        System.out.println();
        media_idade = (double) (idade1 + idade2) / 2;
        System.out.println("A idade media de " + nome1 + " e " + nome2 + " eh de " + media_idade + " anos.");

        sc.close();
    }   
}
